torch/_dynamo/allowed_functions.py

```diff
@@ -307,7 +307,40 @@ def _builtin_constant_ids():
     return rv
 
 
+# A subcheck of is_allowed, we utilize this for patching is_allowed around distributed.
+# We do this because we want to allow these to be traced, and hence covered in skipfiles, but we do not want them to
+# become TorchVariable
```

Comment:
If you don't want to make them as ```TorchVariable```, we should not make them ```is_allowed``` returning True after my refactor. If you just want to inline these functions, can you treat them as regular python functions and go through the regular inline rules?

Reply:
We need some to go one way, some the other, the logic here is correct - and they shouldn't go through inline rules.

Reply:
The problem is they are under torch.* so they get routed to TorchVariable - for some like

```
 if obj in [
            torch.distributed._functional_collectives_impl._all_gather_into_tensor,
            torch.distributed._functional_collectives_impl._all_reduce,
            torch.distributed._functional_collectives_impl._reduce_scatter_tensor,
            torch.distributed._functional_collectives_impl._all_reduce_coalesced,
            torch.distributed._functional_collectives_impl._all_gather_into_tensor_coalesced,
            torch.distributed._functional_collectives_impl._reduce_scatter_tensor_coalesced,
        ]:
```
It is *correct* for them to become TorchVariable

For others (the rest of the checks) - we need to make sure they *do not* become TorchVariable. If we don't have this function, the wrong types will become TorchVariable instead of passing through `CollectiveFunctionRewriteVariable.can_rewrite(value)` `CollectiveFunctionRewriteVariable.can_rewrite(value)` etc

Reply:
I don't quite understand Voz's explanation, but I certainly agree with Voz that there is something funny going on here.

To give an alternate example, on main, I'd like to inline into functions in `_functional_collectives`. I applied this patch:

```
diff --git a/torch/_dynamo/allowed_functions.py b/torch/_dynamo/allowed_functions.py
index 8beca2b4502..d155023ef54 100644
--- a/torch/_dynamo/allowed_functions.py
+++ b/torch/_dynamo/allowed_functions.py
@@ -175,6 +175,7 @@ def _allowed_function_ids():
             # issues observed in
             # https://github.com/pytorch/pytorch/issues/108269
             "torch.distributed.algorithms.",
+            "torch.distributed._functional_collectives.",
         )
         allowed_modules_dot = tuple([x + "." for x in allowed_modules])
         module = inspect.getmodule(obj)
```

But it does not work; for some reason I appear to still be trying to place things like `all_to_all_single` directly into the graph. With Voz's PR and my patch, and turning on `trace_distributed`, only then do I get the expected behavior of inlining.

Reply:
more of a meta question but do we have a design doc for "the new skipfiles/allowed_functions design"? I am a little wary of all this complexity around distributed, hopefully we can do it more cleanly in a redesign